### PR TITLE
JM - Added Incremental Pitest Support

### LINKED ISSUE #24: Feature - Add incremental pitest to this project
# Overview
Add incremental pitest to this project.

# Acceptance Crtierion
Make it possible to run `mvn pitest:mutationCoverage` on just a single class, or a single package, instead of the entire backend code base.

### PR DESCRIPTION
**Pull request by Jumshim**
## Overview
In this PR, I've added support for incremental pitest coverage. Now, developers for this repository can type (for example):

`mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.happiercows.controllers.UserInfoController`

or:

`mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.happiercows.controllers.*`

to selectively run pitest on a class / package.

**NOTE**: Dokku deployment is not necessary to test this PR, thus a link will not be provided.

## Validation (Optional)

1. Download this branch.
2. Run any of the pitest commands above in the root directorty
3. Verify that only the specified package/class is tested.

## Tests

- [X] Backend Unit tests (`mvn test`) pass
- [X] Backend Test coverage (`mvn test jacoco:report`) 100%
- [X] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 

## Linked Issues

Closes #24

### CODE REVIEW

**Review by niyati-m — DISMISSED** [review]
LGTM:)

File: README.md [hunk 1]
```diff
@@ -123,3 +119,25 @@ In order to run the end-to-end tests 'not headless' use the following instead of
...
+For example, to run pitest on just one class, use:
+
+```
+mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.example.controllers.RestaurantsController
+```
+
+To run pitest on just one package, use:
+
+```
+mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.example.controllers.*
```
Comment (pconrad) [inline README.md]:
```suggestion
mvn pitest:mutationCoverage -DtargetClasses=edu.ucsb.cs156.example.controllers.\*
```

That \ is crucial!

Comment (Jumshim) [reply README.md]: Got it!!

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**